// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is govered by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto3";


package monorail;


service Issues {
  rpc CreateIssue (CreateIssueRequest) returns (Issue) {}
  rpc DeleteIssueComment (DeleteIssueCommentRequest) returns (Comment) {}
}


message CreateIssueRequest {
  string project_name = 1;
  Issue issue = 2;
}


message Issue {
  string project_name = 1;
  int64 local_id = 2;
  string summary = 3;
  string description = 4;
}


message DeleteIssueCommentRequest {
  string project_name = 1;
  int64 local_id = 2;
  int64 comment_id = 3;
  bool delete = 4;
}


message Comment {
  string content = 1;
  bool deleted = 2;
}
